pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -72,6 +75,52 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 		numProcPerNode = ptr.Deref(trainJob.Spec.Trainer.NumProcPerNode, intstr.FromString("auto"))
 	}
 
+	// Cap nproc_per_node based on CPU resources when set to "auto" and no GPU is requested
+	if numProcPerNode.String() == "auto" && trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.ResourcesPerNode != nil {
+		// Check if GPU is requested
+		hasGPU := false
+		for resourceName := range trainJob.Spec.Trainer.ResourcesPerNode.Limits {
+			if strings.Contains(strings.ToLower(string(resourceName)), "gpu") {
+				hasGPU = true
+				break
+			}
+		}
+		if !hasGPU {
+			for resourceName := range trainJob.Spec.Trainer.ResourcesPerNode.Requests {
+				if strings.Contains(strings.ToLower(string(resourceName)), "gpu") {
+					hasGPU = true
+					break
+				}
+			}
+		}
+
+		// If no GPU is requested, use CPU limit or default to 1
+		if !hasGPU {
```

Comment:
@tenzen-y @astefanutti Is there any helper function in `k8s.io/utils` we can use to check whether resource (e.g. `cpu`, `gpu`) exists in container.resources ?

Reply:
AFAIK, there is not such util.